src/m365/spo/commands/file/file-sharinglink-add.ts

```diff
@@ -0,0 +1,167 @@
+import { GraphFileDetails } from './GraphFileDetails';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  webUrl: string;
+  fileId?: string;
+  fileUrl?: string;
+  type: string;
+  expirationDateTime: string;
+  scope?: string;
+}
+
+class SpoFileSharingLinkAddCommand extends SpoCommand {
+  private static readonly types: string[] = ['view', 'edit'];
+  private static readonly scopes: string[] = ['anonymous', 'organization'];
+
+  public get name(): string {
+    return commands.FILE_SHARINGLINK_ADD;
+  }
+
+  public get description(): string {
+    return 'Creates a new sharing link for a file';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+    this.#initOptionSets();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        fileId: typeof args.options.fileId !== 'undefined',
+        fileUrl: typeof args.options.fileUrl !== 'undefined',
+        expirationDateTime: typeof args.options.expirationDateTime !== 'undefined',
+        scope: typeof args.options.scope !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-u, --webUrl <webUrl>'
+      },
+      {
+        option: '-i, --fileId [fileId]'
+      },
+      {
+        option: '-f, --fileUrl [fileUrl]'
+      },
+      {
+        option: '--type <type>',
+        autocomplete: SpoFileSharingLinkAddCommand.types
+      },
+      {
+        option: '--expirationDateTime [expirationDateTime]'
+      },
+      {
+        option: '--scope [scope]',
+        autocomplete: SpoFileSharingLinkAddCommand.scopes
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        const isValidSharePointUrl: boolean | string = validation.isValidSharePointUrl(args.options.webUrl);
+        if (isValidSharePointUrl !== true) {
+          return isValidSharePointUrl;
+        }
+
+        if (args.options.fileId && !validation.isValidGuid(args.options.fileId)) {
+          return `${args.options.fileId} is not a valid GUID`;
+        }
+
+        if (args.options.type &&
+          SpoFileSharingLinkAddCommand.types.indexOf(args.options.type) < 0) {
+          return `'${args.options.type}' is not a valid type. Allowed types are ${SpoFileSharingLinkAddCommand.types.join(', ')}`;
+        }
+
+        if (args.options.scope &&
+          SpoFileSharingLinkAddCommand.scopes.indexOf(args.options.scope) < 0) {
+          return `'${args.options.scope}' is not a valid scope. Allowed scopes are ${SpoFileSharingLinkAddCommand.scopes.join(', ')}`;
+        }
+
+        if (args.options.expirationDateTime && !validation.isValidISODateTime(args.options.expirationDateTime)) {
+          return `${args.options.expirationDateTime} is not a valid ISO date string.`;
+        }
+
```

Comment:
expirationDateTime only works for anonymous links. let's ensure this in the validation that `expirationDateTime` only works when the scope is `anonymous`. Let's also mention this in the docs in the description of `expirationDateTime`.

Update: if we don't provide a `scope` value, the validation should pass as well because we don't know if expirationDateTime is the default link type or not.